### PR TITLE
Update README with basic usage instructions.

### PR DESCRIPTION
**Pull request by jsuchome**
We need some examples descriptions, but let's wait for https://github.com/fuseml/fuseml-core/pull/43

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,8 +1,251 @@
...
+    ```
+    This will produce server (`bin/fuseml_core`) and command line client (`bin/fuseml_core-cli*`) binaries. The client binaries are by default compiled for multiple architectures, so pick one you need according to your OS. For convenience, you might create a link to the binary with more reasonable name, like
+
+    ```bash
+    ln -s bin/fuseml_core-cli-linux-amd64 bin/fuseml_core-cli
+    ```
+
+## Usage
+
+* Run the server localy
```
Comment (flaviodsr) [inline README.md:49]: I am not sure about offering the option to run `fuseml_core` outside of the environment provided by `fuseml-installer`. If we go that way we should also mention everything that is required to be able to run it properly (gitea, tekton, istio, ...), not only the environment variables.

IMO we should simple not offer that.

File: README.md [hunk 2]
```diff
@@ -1,8 +1,251 @@
...
+    bin/fuseml_core-cli application list --type "predictor" --workflow "mlflow-sklearn-e2e"
+    bin/fuseml_core-cli codeset list --project "mlflow-project-01" --label "mlflow"
+    bin/fuseml_core-cli runnable list --id "ml-trainer-123" --kind "trainer" --labels '{
+          "function": "predict|train",
+          "library": "pytorch"
+       }'
+    bin/fuseml_core-cli workflow list --name "workflow"
+```
+
+  Without `-url` argument, client tries to connect to the server running locally at your machine. To find the URL of the fuseml-core server service that is installed in your Kubernetes cluster, call
```
Comment (flaviodsr) [inline README.md:105]: We need to fix this, having to pass url is not really user friendly.

File: README.md [hunk 3]
```diff
@@ -1,8 +1,251 @@
...
+
+  From the usage output you can see that with the client you can operate with various FuseML artefacts.
+
+  * Codesets describe the code of your ML application, for example MLflow project. They are currently implemented as git repositories.
+
+    Create new codeset by calling `bin/fuseml_core-cli codeset register` command. use the `--help` option to get the description of the command usage.
+
+    Example:
+    ```bash
+    bin/fuseml_core-cli codeset register -name "test" --project "mlflow-project-01" --location "/tmp/mlflow/mlflow-01"
```
Comment (flaviodsr) [inline README.md:121]: why some parameter are passed with `-` and others with `--`?

File: README.md [hunk 4]
```diff
@@ -1,8 +1,251 @@
...
+    bin/fuseml_core-cli workflow register
+    ``` command.
+
+    After the workflow is registered, you should assign a codeset to it. That way the workflow will be automatically executed every time you push a new change to your git repository that is represented by a codeset.
+
+    ```bash
+    bin/fuseml_core-cli workflow assign -name "workflow-name" -codeset-name "test" -codeset-project "mlflow-project-01"
+    ```
+
+    To see the progress of running workflow, check the Tekton dashboard in your browser (remember `TEKTON_DASHBOARD_URL` extracted earlier).
```
Comment (flaviodsr) [inline README.md:148]: or use `workflow list-runs`?

File: README.md [hunk 5]
```diff
@@ -1,8 +1,251 @@
...
+
+    For registering new workflow, use
+    ```bash
+    bin/fuseml_core-cli workflow register
+    ``` command.
+
+    After the workflow is registered, you should assign a codeset to it. That way the workflow will be automatically executed every time you push a new change to your git repository that is represented by a codeset.
+
+    ```bash
+    bin/fuseml_core-cli workflow assign -name "workflow-name" -codeset-name "test" -codeset-project "mlflow-project-01"
```
Comment (flaviodsr) [inline README.md:145]: might be worthy mentioning that when assigning a workflow to a codeset, the workflow is also triggered.

File: README.md [hunk 6]
```diff
@@ -1,8 +1,251 @@
...
+
+  Now replace the original values in the pipeline-01.yaml example. You can do it by editing the file manually or by running following command:
+  ```bash
+  sed -i -e "/AWS_ACCESS_KEY_ID/{N;s/value: [^ \t]*/value: $ACCESS/}" pipelines/pipeline-01.yaml
+  sed -i -e "/AWS_SECRET_ACCESS_KEY/{N;s/value: [^ \t]*/value: $SECRET/}" pipelines/pipeline-01.yaml
+  ```
+
+* Create a workflow
+
+  Use the modified example workflow definition:a
```
Comment (flaviodsr) [inline README.md:215]: typo: `definition:a`

File: README.md [hunk 7]
```diff
@@ -1,8 +1,251 @@
...
+  fuseml_core-cli workflow register --body "$wf"
+  ```
+
+* Assign the codeset to workflow
+
+  ```bash
+  fuseml_core-cli workflow assign -name mlflow-sklearn-e2e -codeset-name mlflow-test -codeset-project mlflow-project-01
+  ```
+
+  Now that the Worfklow got specific Codeset assigned, it initiated workflow run.
```
Comment (flaviodsr) [inline README.md:229]: "...initiated `a` workflow run"?

File: README.md [hunk 8]
```diff
@@ -1,8 +1,251 @@
...
+  ```
+
+* Assign the codeset to workflow
+
+  ```bash
+  fuseml_core-cli workflow assign -name mlflow-sklearn-e2e -codeset-name mlflow-test -codeset-project mlflow-project-01
+  ```
+
+  Now that the Worfklow got specific Codeset assigned, it initiated workflow run. 
+  Browse to `TEKTON_DASHBOARD_URL` and click PipelineRuns. Once the run is succeeded, new application will occur.
```
Comment (flaviodsr) [inline README.md:230]: or use `workflow list-runs`?

I would also change "new application will occur" to "a new FuseML application will be created".

File: README.md [hunk 9]
```diff
@@ -1,8 +1,251 @@
...
+
+* Use the prediction service
+
+  Check the applications list with
+
+  ```bash
+  fuseml_core-cli application list
+  ```
+  
+  Use the URL from the new application to run the prediction.
```
Comment (flaviodsr) [inline README.md:240]: might be good to also add an example `curl` command to perform the prediction.

**Review by flaviodsr — CHANGES_REQUESTED** [review]
Looks very nice, just some suggestions!

File: README.md [hunk 10]
```diff
@@ -1,8 +1,251 @@
...
+  * Install go dependencies
+    ```bash
+    make deps
+    ```
+
+  * Generate server, client and CLI code and build the binaries.
+    ```bash
+    make all
+    ```
+    This will produce server (`bin/fuseml_core`) and command line client (`bin/fuseml_core-cli*`) binaries. The client binaries are by default compiled for multiple architectures, so pick one you need according to your OS. For convenience, you might create a link to the binary with more reasonable name, like
```
[the remark this answers is not in the document]
Comment (jsuchome) [reply README.md:41]: Right, and the symlink is no longer necessary.

File: README.md (hunk 1 above)
Comment (jsuchome) [reply README.md:49]: This is not official documentation, this is a README file, which should serve developers too. I agree that we should omit such option from official documentation, but here it makes sense.

Comment (jsuchome) [reply README.md:49]: I'll update the wording to reflect the situation

File: README.md (hunk 2 above)
Comment (jsuchome) [reply README.md:105]: https://github.com/fuseml/fuseml-core/pull/54

File: README.md (hunk 3 above)
Comment (jsuchome) [reply README.md:121]: For some reason, it is supported by goa ... but you are right it should be consistent in the docs...

File: README.md (hunk 5 above)
Comment (jsuchome) [reply README.md:145]: OK, I'll add it to the line above.

File: README.md (hunk 4 above)
Comment (jsuchome) [reply README.md:148]: Good point, I did not play with that yet.

File: README.md (hunk 9 above)
Comment (jsuchome) [reply README.md:240]: Indeed, I just have to dig it from somewhere ...

File: README.md [hunk 11]
```diff
@@ -1,8 +1,313 @@
...
+    make all
+    ```
+    This will produce server (`bin/fuseml_core`) and command line client (`bin/fuseml`) binaries. The client binary is compiled for your current architecture.
+
+
+## Usage
+
+* Run the server localy
+
+  After successful installation using [fuseml-installer](https://github.com/fuseml/fuseml), `fuseml-core` server is already running in the Kubernetes cluster.However, for developmnent and testing purposes it is possible to run it locally on your machine by executing `bin/fuseml_core`. It's necessary to set some environment variables first.
```
Comment (flaviodsr) [inline README.md:48]: Missing space in `...cluster.However...`

`run it locally on your machine` sounds a bit redundant for me.

Suggestion: mention the requirements (setting the env variables) before running the binary, so it follows the sequence. Something like `... its possible to run it locally by setting the following env variables... and executing `bin/fuseml_core`

File: README.md [hunk 12]
```diff
@@ -1,8 +1,313 @@
...
+    bin/fuseml application list --type "predictor" --workflow "mlflow-sklearn-e2e"
+    bin/fuseml codeset list --project "mlflow-project-01" --label "mlflow"
+    bin/fuseml runnable list --id "ml-trainer-123" --kind "trainer" --labels '{
+          "function": "predict|train",
+          "library": "pytorch"
+       }'
+    bin/fuseml workflow list --name "workflow"
+```
+
+  Instead of providing `-url` argument to each command call, save the service URL into `FUSEML_SERVER_URL` enviornment variable and export it. Use this command to fill the variable with the URL of the `fuseml-core` server service that is installed in your Kubernetes cluster:
```
Comment (stefannica) [inline README.md:102]: typo: enviornment

File: README.md [hunk 13]
```diff
@@ -1,8 +1,313 @@
...
+
+  ```bash
+  export FUSEML_SERVER_URL=$(kubectl get VirtualService -n fuseml-core fuseml-core -o jsonpath="{.spec.hosts[0]}")
+  ```
+
+  If neither `FUSEML_SERVER_URL` nor `-url` value is set, client tries to connect to the server running locally at your machine.
+
+
+  From the usage output you can see that with the client you can operate with various FuseML artefacts.
```
Comment (flaviodsr) [inline README.md:111]: The wording here sounds a bit confusing. IMO you do not need to mention the fuseml output. Something like: `The FuseML client allows you to manage the various supported artefacts (application, codeset, runnable and workflow).`

**Review by stefannica — COMMENTED** [review]
Very nice, thanks for taking the time to update it !

File: README.md [hunk 14]
```diff
@@ -1,8 +1,313 @@
...
+
+  * Codesets describe the code of your ML application, for example MLflow project. They are currently implemented as git repositories.
```
Comment (flaviodsr) [inline README.md:113]: What does `describe the code` means?

File: README.md [hunk 15]
```diff
@@ -1,8 +1,313 @@
...
+
+    Create new codeset by calling `bin/fuseml codeset register` command. use the `--help` option to get the description of the command usage.
```
Comment (flaviodsr) [inline README.md:115]: `... a new codeset by calling the`?

Regarding the help it might be better to just mention at the beginning of this section that you can use `--help` for each command to get more details.

File: README.md [hunk 16]
```diff
@@ -1,8 +1,313 @@
...
+    bin/fuseml codeset register --name "test" --project "mlflow-project-01" --location "/tmp/mlflow/mlflow-01"
+    ```
+
+    `--location` argument points to the directory on your machine where your ML application code is located.
+
+    After registering, use
+    ```
+    bin/fuseml codeset list
+    ```
+    command to list available codesets and check the value of "URL" in the output of your newly registered codeset. Use this value to git clone the code into another directory. Now you can work in this directory just like with any other project saved in git. Assuming you have configured a workflow assignment (see parts about workflows), any time you commit and push new changes to your code, a worklflow run will be executed and you can expect updated application as a result.
```
Comment (flaviodsr) [inline README.md:128]: suggestion: `...Assuming that you assigned a workflow to this codeset (see workflows section), every time you push new code changes a new workflow run will be created. `

File: README.md [hunk 17]
```diff
@@ -1,8 +1,313 @@
...
+  fuseml workflow register --body "$(cat pipelines/pipeline-01.yaml)"
+  ```
+
+* Assign the codeset to workflow
+
+  ```bash
+  fuseml workflow assign --name mlflow-sklearn-e2e --codeset-name mlflow-test --codeset-project mlflow-project-01
+  ```
+
+  Now that the Worfklow got specific Codeset assigned, it has initiated a workflow run. To watch the workflow progress, check "workflow run" with
```
Comment (flaviodsr) [inline README.md:231]: `Now that the Workflow is assigned to the Codeset, a new workflow run was created.`

**Review by flaviodsr — CHANGES_REQUESTED** [review]

File: README.md (hunk 11 above)
Comment (jsuchome) [reply README.md:48]: Good idea.

File: README.md (hunk 13 above)
Comment (jsuchome) [reply README.md:111]: ok

File: README.md (hunk 14 above)
Comment (jsuchome) [reply README.md:113]: OK, that might look confusing. What about "Codesets contain the code"?

File: README.md (hunk 15 above)
Comment (jsuchome) [reply README.md:115]: ok

File: README.md (hunk 14 above)
Comment (flaviodsr) [reply README.md]: sounds good to me

**Review by flaviodsr — APPROVED** [review]
